l10n_co_geo/models/res_company.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+###############################################################################
+#    License, author and contributors information in:                         #
+#    __openerp__.py file at the root folder of this module.                   #
+###############################################################################
+
+from openerp import models, fields, api
+
+
+class ResCompany(models.Model):
+    _name = 'res.company'
+    _inherit = 'res.company'
+
+    country_id = fields.Many2one(
+        comodel_name='res.country',
+        related='city_id.state_id.country_id',
+        readonly=True,
+        required=False,
+        index=False,
+        default=None,
+        help=False,
+        domain=[],
+        context={},
+        ondelete='cascade',
+        auto_join=False
+    )
+
+    city = fields.Char(
+        invisible=True
+    )
+
+    state_id = fields.Many2one(
+        'res.country.state',
+        related='city_id.state_id',
+        readonly=True,
+    )
+
+    city_id = fields.Many2one(
```

Comment:
As the author I can relicense it in it's current state, nobody can force me to only license it under AGPL only, so juristically speaking, we can cleanly separate concerns. Obviously, once someone else gains co-authorship we would need to maintain a fork and cannot use adaptations. Now I think strictly speaking we do not depend on the code, rather on the data the code, with checks like "if this client is habitant of this city, then do something", but I think we have a problem as well, because we would need to release the data under AGPL, too, because obviously it's dependent on the code, and in consequence, everything that depends on the data needs to be made available under AGPL too. This is very problematic as we need to protect our "inventions" at least for some time...